examples/put_to_self_logical.c

```diff
@@ -0,0 +1,189 @@
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlNIInit failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto fini;
+	}
+
+	ret = PtlEQAlloc( nih, 10, & eqh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto ni_fini;
+	}
+
+	ret = PtlPTAlloc( nih, 0,  eqh, PTL_PT_ANY, &pti);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlPTAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto eq_free;
+	}
+
+	ret = PtlGetPhysId(nih, &id);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlGetId failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto pt_free;
+	} else {
+		mapping[0] = id;
+	}
+
+	/* set the mapping from our logical id (rank) to our physical id (nid and pid)*/
+	ret = PtlSetMap( nih, 1, mapping);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlSetMap failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto pt_free;
+	}
+
+	/*describe the memory used to receive the message*/
+	le = (ptl_le_t){ .start = receive_data,
+			.length = sizeof(data),
+			.ct_handle = PTL_CT_NONE,
+			.uid = PTL_UID_ANY,
+			.options = PTL_LE_OP_PUT };
+
+	/* expose to the network the receive buffer */
+	ret = PtlLEAppend( nih,  pti, &le, PTL_PRIORITY_LIST, NULL, & leh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlLEAppend failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto pt_free;
+	}
+
+	/* wait for the LINK event that confirms we are ready for message reception*/
+	ret = PtlEQWait( eqh, & ev);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto le_unlink;
+	}
+	PtlEvToStr(0, & ev, msg);
+	printf("Event : %s\n", msg);
+
+	/*describe the memory used to send the message*/
+	md = (ptl_md_t){ .start = data,
+			.length = sizeof(data),
+			.options = 0,
+			.eq_handle = eqh,
+			.ct_handle = PTL_CT_NONE };
+
+	/* create a memory descriptor to be used to send message */
+	ret = PtlMDBind( nih, &md, & mdh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlMDBind failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto le_unlink;
+	}
+
+	/*send the message to ourself*/
+	ret = PtlPut( mdh, 0, sizeof(data), PTL_ACK_REQ, rank0,  pti, 0, 0, NULL, 0);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlPut failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto md_release;
+	}
+
+	/*wait to receive the send event message*/
+	ret = PtlEQWait( eqh, & ev);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto md_release;
+	}
+	PtlEvToStr(0, & ev, msg);
+	printf("Event : %s\n", msg);
+
+	/*wait to receive the ack from our recipent*/
+	ret = PtlEQWait( eqh, & ev);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto md_release;
+	}
+	PtlEvToStr(0, & ev, msg);
+	printf("Event : %s\n", msg);
+
+	ret = PtlEQWait( eqh, & ev);

```

Comment:
should rather use a loop to retrieve the three events since there they can arrive in any order (like in put_to_self.c)